src/OpenTelemetry.Instrumentation.Runtime/Options/GcMetricOptions.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="GcMetricOptions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Instrumentation.Runtime.Options
+{
+    /// <summary>
+    /// Enum for garbage collection metrics.
+    /// </summary>
+    [Flags]
+    public enum GcMetricOptions : int
+    {
+        /// <summary>
+        /// Do not set flag for any metrics.
+        /// </summary>
+        None = 0x00,
+
+        /// <summary>
+        /// Set flag for all options.
+        /// </summary>
+        All = Count
+#if NETCOREAPP3_1_OR_GREATER
+            | AllocatedBytes
+#endif
+
+#if NET6_0_OR_GREATER
+            | CommitedBytes | HeapSize | FragmentationSize
+#endif
+            ,
```

Comment:
Warning due to space before comma. Removing the space doesn't work either. Might have to define this enum option for each frameworks with `#if.. #elseif...#else...#endif`.

Reply:
Can't use "else if" because when it's `NET6_0_OR_GREATER`, `NETCOREAPP3_1_OR_GREATER` has to be enabled as well. Suppressed it with `#pragma warning disable SA1001`.